internal/compiler-bridge/src/main/scala/xsbt/Dependency.scala

```diff
@@ -275,21 +275,21 @@ final class Dependency(val global: CallbackGlobal) extends LocateClassFile with
         }
       }
 
-      val cache = scala.collection.mutable.Map.empty[Symbol, (Handler, scala.collection.mutable.HashSet[Type])]
+      val cache = new util.HashMap[Symbol, (Handler, util.HashSet[Type])]()
       private var handler: Handler = _
       private var visitedOwner: Symbol = _
       def setOwner(owner: Symbol) = {
         if (visitedOwner != owner) {
-          cache.get(owner) match {
-            case Some((h, ts)) =>
-              visited = ts
-              handler = h
-            case None =>
-              val newVisited = scala.collection.mutable.HashSet.empty[Type]
-              handler = createHandler(owner)
-              cache += owner -> (handler -> newVisited)
-              visited = newVisited
-              visitedOwner = owner
+          if (cache.containsKey(owner)) {
```

Comment:
can always call get and check if this is not a null? I know this is ugly but we save multiple some set operation then.